contracts/pm/mixins/MixinReserve.sol

```diff
@@ -0,0 +1,188 @@
+     * @param _amount Amount of funds to add to reserve
+     */
+    function addReserve(address _reserveHolder, uint256 _amount) internal {
+        ReserveManager storage manager = reserveManagers[_reserveHolder];
+        Reserve storage reserve = manager.reserve;
+
+        reserve.fundsAdded = remainingReserve(_reserveHolder).add(_amount);
+
+        // If reserve is thawed then clear unneeded contract storage
+        // for the reserve and update the reserve while ensuring that
+        // any additional funds are added to the remaining funds in the reserve
+        if (reserveState(_reserveHolder) == ReserveState.Thawed) {
+            // We clear these individual fields of the reserve instead
+            // of using `clearReserve()` because we can directly update
+            // fundsAdded with its new value instead of clearing the old
+            // value and then setting its new value
+            reserve.fundsClaimed = 0;
+            reserve.freezeRound = 0;
+            reserve.recipientsInFreezeRound = 0;
+            // Increment reserveNonce so the manager can point to a new
+            // mapping for tracking funds claimed from a reserve by
+            // different claimants
+            manager.reserveNonce = manager.reserveNonce.add(1);
+        }
+
+        emit ReserveFunded(_reserveHolder, _amount);
+    }
+
+    /**
+     * @dev Clears contract storage used for a reserve
+     * @param _reserveHolder Address of reserve holder
+     */
+    function clearReserve(address _reserveHolder) internal {
+        ReserveManager storage manager = reserveManagers[_reserveHolder];
+        delete manager.reserve;
+        // Increment reserveNonce so the manager can point to a new
+        // mapping for tracking funds claimed from a reserve by
+        // different claimants
+        manager.reserveNonce = manager.reserveNonce.add(1);
+    }
+
+    /**
+     * @dev Claims funds from a reserve
+     * @param _reserveHolder Address of reserve holder
+     * @param _claimant Address of claimant
+     * @param _amount Amount of funds to claim from the reserve
+     * @return Amount of funds (<= `_amount`) claimed by `_claimant` from the reserve for `_reserveHolder`
+     */
+    function claimFromReserve(
+        address _reserveHolder,
+        address _claimant,
+        uint256 _amount
+    )
+        internal
+        returns (uint256)
+    {
+        ReserveManager storage manager = reserveManagers[_reserveHolder];
+        Reserve storage reserve = manager.reserve;
+
+        // If reserve is not frozen then freeze it
+        if (reserveState(_reserveHolder) == ReserveState.NotFrozen) {
+            uint256 freezeRound = roundsManager().currentRound();
+            // TODO: make sure bondingManager.getTranscoderPoolSize()
+            // returns the locked in # registered for the freeze round
+            uint256 recipientsInFreezeRound = bondingManager().getTranscoderPoolSize();
```

Comment:
Should we get the size for the entire transcoder pool or the active set here?

Reply:
We want the size of the entire transcoder pool here because the entire transcoder pool will be the active set once we implement the staking protocol changes. As mentioned in the TODO, we'll need to make sure that there is a way to get the locked in transcoder pool size for a round when we're implementing those changes